### ISSUE TITLE
Switch --to-json does not seem to work

### ISSUE THREAD

**Issue by hifzakh** [RedHatInsights/insights-client#81]
According to the documentation, `--to-json` switch lets you print the collection results to the console as JSON, but it seems that the `insights-client` is still using the default option i.e. `archive`.

```
[user@hostname ~]# insights-client --to-json --offline
Starting to collect Insights data for [hostname]
Archive saved at /var/tmp/vcqru0qm/insights-[hostname]-20190703183658.tar.gz
```

**Comment by pipopopo** [comment]
Works for me:
~~~
# insights-client --to-json
Starting to collect Insights data for ktordeur-sat65.sysmgmt.lan
Uploading Insights data.
Successfully uploaded report from xxxxxxxxxx to accountxxxxxx.
{"details": {"sat6_basic_check|SAT6_BASIC_SYSTEM_REQUIREMENT_LOST": {"requirement": {"osver_min": {"7": 0},
...
~~~

What version of insights are you running?
# insights-client --version

**Comment by hifzakh** [comment]
I am using a RHEL8 box.
```
insights-client --version
Client: 3.0.5-4
Core: 3.0.102-1
``

**Comment by hifzakh** [comment]
May be you can try running the command with the `--offline` switch to reproduce the issue.

**Comment by hifzakh** [comment]
I tried out a couple of cases and figured out that `--to-json` depends upon the status of the system. It works if the system is registered, otherwise not. I do not think that the format of the output should have anything to do with the registration of the system.

**Comment by pipopopo** [comment]
@hifzakh If I'm not mistaken part of the --to-json is the collection of the rules that were found for the system in question, as the rules can only be run on cloud.redhat.com and not locally it makes sense it requires the system to be registered.

~~~
# insights-client --to-json
Starting to collect Insights data for ktordeur-sat65.sysmgmt.lan
Uploading Insights data.
Successfully uploaded report from xxxxxxxxxxx to account xxxx.
{"details": {"sat6_basic_check|SAT6_BASIC_SYSTEM_REQUIREMENT_LOST": {"requirement": {"osver_min": {"7": 0}, "cpu_core": 4, "swap_size": 4, "ram_size": 20}, "error_key": "SAT6_BASIC_SYSTEM_REQUIREMENT_LOST", "type": "rule", "satver": [6, 5], "mem_swap": {"mem": 14.18}}, "CVE_2017_5753_4_cpu_kernel|KERNEL_CVE_2017_5753_4_CPU_ERROR_3": {"cves_pass": ["CVE-2017-5753", "CVE-2017-5754"], "debugfs_available": true, "dmesg_available": true, "retpo_kernel_but_no_sys_cpu_vuln": true, "package_name": "kernel", "running_kernel": "3.10.0-957.21.3.el7.x86_64", "sysfs_vuln_s1": null, "dmesg_wrapped": false, "problems": {"v2_vulnerable": true, 
...
~~~

**Comment by portante** [comment]
@pipopopo, can you help me understand this better?

The `insights-client` offers two kinds of output, `--to-json`, and the default behavior of generating an archive.  Is that the case?  Or is it the case that `--to-json` is directing the client to do something different than when the client is run without `--to-json`?

We are interested in finding out the exact contents of _what would be uploaded_, whether or not the system is registered or not, in JSON form.

So for a registered system, we'd like the following to work:

```
$ insights-client --to-json > what-would-be-uploaded.json
```

And for an unregistered system:

```
$ insights-client --to-json --offline > what-would-be-uploaded.json
```

Does that make sense?

We work on some systems that are registered, and others that are not registered.  We'd like to be able to collect as much data from `insights-client` as we can in both cases.

**Comment by lphiri** [comment]
@portante  --to-json only applies to registered systems - its meant to dump the results of the insights scan as returned from the Insights service after you post data.  There is currently no support for dumping client side collected data in json format.

**Comment by portante** [comment]
@lphiri, thanks that explains it.

**Comment by gravitypriest** [comment]
Adding strict disallowance of --to-json combined with --offline:
https://github.com/RedHatInsights/insights-core/pull/2106

**Comment by gravitypriest** [comment]
PR was merged and released as part of core 3.0.115. Marking as resolved.
